### PR TITLE
fix: csharp generator not render optional types for optional properties

### LINKED ISSUE #941: C# generator does not render optional properties correctly
#### Describe the bug
As seen in the following example, the C# generator does not render optional properties `string?` as it does on master:
https://github.com/asyncapi/modelina/blob/next/examples/csharp-generate-required-properties/__snapshots__/index.spec.ts.snap
https://github.com/asyncapi/modelina/blob/master/examples/csharp-generate-required-properties/__snapshots__/index.spec.ts.snap

### PR DESCRIPTION
**Pull request by zaytsevand**
**Description**

- Using ConstrainedObjectPropertyModel to get info on if the property is required

**Related issue(s)**
Fixes #941

### CODE REVIEW

**Comment by zaytsevand** [comment]
@jonaslagoni I don't get what's wrong here, but black-box tests are constantly failing

File: src/generators/csharp/CSharpConstrainer.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
+import { ConstrainedObjectPropertyModel } from 'models';
 import { ConstrainedEnumValueModel } from '../../models';
```
Comment (jonaslagoni) [inline src/generators/csharp/CSharpConstrainer.ts:2]:
```suggestion
import { ConstrainedEnumValueModel, ConstrainedObjectPropertyModel } from '../../models';
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
My comment fixes your blackbox tests and the linter you can probably fix with `npm run lint:fix`

**Review by jonaslagoni — APPROVED** [review]
LGTM 🔥

**Pull request merged**